libcontainer/setns_init_linux.go

```diff
@@ -65,6 +66,15 @@ func (l *linuxSetnsInit) Init() error {
 		unix.Umask(int(*l.config.Config.Umask))
 	}
 
+	if l.config.Config.Scheduler != nil {
+		if err := unix.SchedSetAttr(0, utils.ToSchedAttr(l.config.Config.Scheduler), 0); err != nil {
+			if errors.Is(err, unix.EPERM) {
+				return fmt.Errorf("error setting scheduler(please check you have appropriate privileges or valid cpus config): %w", err)
```

Comment:
If you do want to distinguish between the two cases for EPERM, maybe it makes sense to check if we have `cpus` set first.

Also, maybe move this code into a separate function (setupScheduler), as we have the same 8 lines of code in both setns_init_linux.go and standard_init_linux.go

Reply:
> maybe it makes sense to check if we have `cpus` set first.

I also considered it in the right time, but there is another corner case is that if the `cpus` value includes all cpus in the machine, it will be no error. So I don't want to check cpus value.